### PR TITLE
Tf custom estimator

### PR DESCRIPTION
**Pull request by nirtiac**
Very very rough, leaves a lot to be implemented, code needs to be cleaned and commented properly. Training is currently absurdly slow so not yet tested. Creating a pull request simply so that we can see the code structure. Remarks:
- very awkward creation of the estimator model function; parameter passing is confusing. 
- I had to do some hacking to get the train data into the format expected; the tensorflow estimator class is definitely meant to be used with their dataset class.

### CODE REVIEW

File: vulcanai/net.py [hunk 1]
```diff
@@ -556,72 +351,47 @@ def train(self, epochs, train_x, train_y, val_x, val_y,
...
                 epoch_time = time.time()
                 print("--> Epoch: {}/{}".format(
-                    epoch,
-                    epochs - 1
+                        epoch,
+                        epochs - 1
                 ))
 
+                #TODO: I think I still need to leave this in?
                 train_x, train_y = shuffle(train_x, train_y, random_state=0)
```
Comment (rfratila) [inline vulcanai/net.py]: It's good to shuffle the training set before training

File: vulcanai/net.py [hunk 2]
```diff
@@ -556,72 +351,47 @@ def train(self, epochs, train_x, train_y, val_x, val_y,
...
+
+                train_eval_result = self.classifier.evaluate(input_fn=lambda:self.eval_input_fn(train_x_dict, train_y, batch_ratio))
+                val_eval_result = self.classifier.evaluate(input_fn=lambda:self.eval_input_fn(val_x_dict, val_y, batch_ratio))
+
+                train_error = train_eval_result["loss"]
+                train_accuracy = train_eval_result["accuracy"]
+                validation_error = val_eval_result["loss"]
+                validation_accuracy = val_eval_result['accuracy']
 
                 if self.stopping_rule == 'best_validation_error' and validation_error < best_error:
```
Comment (rfratila) [inline vulcanai/net.py]: See if tf.estimators has any early stopping rule functionality

File: vulcanai/ops.py [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
-    "softmax": softmax,
-    "rectify": rectify,
-    "selu": selu
+    "sigmoid": tf.nn.sigmoid,
+    "softmax": tf.nn.softmax,
+    "rectify": tf.nn.relu, #TODO: this is it right?
+    "selu": tf.nn.selu
 }
 
+#TODO: kinda dumb
```
Comment (rfratila) [inline vulcanai/ops.py]: should we just default it to Adam and get rid of this option all together?
